src/display/annotation_layer.js

```diff
@@ -305,127 +306,35 @@ var TextAnnotationElement = (function TextAnnotationElementClosure() {
      * @returns {HTMLSectionElement}
      */
     render: function TextAnnotationElement_render() {
-      var rect = this.data.rect, container = this.container;
-
-      // Sanity check because of OOo-generated PDFs.
```

Comment:
Before PR #6770, [see this code](https://github.com/mozilla/pdf.js/pull/6770/files#diff-1097e3b131ba22f8e8a6c2f20e3ab5ecL175), the `rect` was potentially modified before getting the `container`. However, given that the old code seems to yield the *exact* same annotation display behaviour as with this patch (using your test-case), removing it seems fine to me.
Thanks for the in-depth explanation!